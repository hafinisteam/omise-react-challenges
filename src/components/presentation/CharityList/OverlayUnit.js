import React from 'react';
import { animated } from 'react-spring';
import { Radio, Popconfirm, Select } from 'antd';
import styled from 'styled-components';
import { UIButton } from '~/components/presentation/ui/button';
import { donateRange } from '~/configs';
import currencyData from '~/configs/currency.json';
import { mediaMax } from '~/utilities/layout';

const { Option } = Select;

export const OverlayWrapper = styled(animated.div)`
	position: absolute;
	left: 0;
	top: 0;
	width: 100%;
	height: 100%;
	display: flex;
	justify-content: center;
	align-items: center;
	flex-direction: column;
	background: rgba(255, 255, 255, 0.9);
	text-align: center;
	.close-button {
		font-size: 18px;
		font-weight: 700;
		position: absolute;
		right: 10px;
		top: 10px;
		border: none;
		background: transparent;
		cursor: pointer;
	}
	.intro {
		font-size: 18px;
		font-weight: 700;
		margin-bottom: 15px;
		span {
			display: inline-block;
			margin-right: 10px;
		}
	}
	.range {
		margin-bottom: 20px;
	}
	span.ant-radio + * {
		font-size: 16px;
		font-weight: 700;
	}
	${mediaMax.xl`
		.intro {
			font-size: 16px;
			span {
				margin-bottom: 10px;
			}
		}
		span.ant-radio + * {
			font-size: 14px;
			font-weight: 400;
		}
	`}
`;

const OverlayUnit = props => {
	return (
		<div>
			<button className="close-button" onClick={props.onCloseOverlay}>
				x
			</button>
			<div className="intro">
				<span>Select the amount to donate</span>
				<Select
					defaultValue={props.currency}
					style={{ width: 170 }}
					onChange={props.onSelect}
				>
					{Object.keys(currencyData).map(k => (
						<Option value={currencyData[k].code} key={k}>
							{currencyData[k].name}
						</Option>
					))}
				</Select>
			</div>
			<div className="range">
				<Radio.Group onChange={props.onChangeRadio} value={props.value}>
					{donateRange.map(d => (
						<Radio value={d} key={d}>
							{d}
						</Radio>
					))}
				</Radio.Group>
			</div>
			<Popconfirm
				title="Are you sure ?"
				onConfirm={props.onPayDonation}
				okText="Yes"
				cancelText="No"
			>
				<UIButton>Pay</UIButton>
			</Popconfirm>
		</div>
	);
};

export default OverlayUnit;
