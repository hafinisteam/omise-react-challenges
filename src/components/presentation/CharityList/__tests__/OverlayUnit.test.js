import "jest-styled-components";
import renderer from "react-test-renderer";
import OverlayUnit from "../OverlayUnit";
import sinon from "sinon";

describe("<OverlayUnit />", () => {
  it("should render correctly", () => {
    const tree = renderer.create(<OverlayUnit />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("close button should call click events", () => {
    const onButonClick = sinon.spy();
    const wrapper = shallow(<OverlayUnit onCloseOverlay={onButonClick} />);
    wrapper.find("button.close-button").simulate("click");
    expect(onButonClick).toHaveProperty("callCount", 1);
  });
});
