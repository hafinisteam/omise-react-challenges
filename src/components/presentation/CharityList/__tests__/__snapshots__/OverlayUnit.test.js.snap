// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OverlayUnit /> should render correctly 1`] = `
.c0 {
  border: 1px solid #4674EE;
  border-radius: 3px;
  background: white;
  color: #4674EE;
  font-size: 16px;
  padding: 5px 10px;
  cursor: pointer;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  font-weight: 600;
}

.c0:hover {
  color: white;
  background: #4674EE;
}

<div>
  <button
    className="close-button"
    onClick={[Function]}
  >
    x
  </button>
  <div
    className="intro"
  >
    <span>
      Select the amount to donate
    </span>
    <div
      className="ant-select ant-select-enabled"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onMouseOut={[Function]}
      onMouseUp={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "width": 170,
        }
      }
    >
      <div
        aria-autocomplete="list"
        aria-controls="test-uuid"
        aria-expanded={false}
        aria-haspopup="true"
        className="ant-select-selection
            ant-select-selection--single"
        onKeyDown={[Function]}
        role="combobox"
        tabIndex={0}
      >
        <div
          className="ant-select-selection__rendered"
        >
          <div
            className="ant-select-selection-selected-value"
            style={
              Object {
                "display": "block",
                "opacity": 1,
              }
            }
            title="US Dollar"
          >
            US Dollar
          </div>
        </div>
        <span
          className="ant-select-arrow"
          onClick={[Function]}
          style={
            Object {
              "WebkitUserSelect": "none",
              "userSelect": "none",
            }
          }
          unselectable="on"
        >
          <i
            aria-label="icon: down"
            className="anticon anticon-down ant-select-arrow-icon"
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="down"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
              />
            </svg>
          </i>
        </span>
      </div>
    </div>
  </div>
  <div
    className="range"
  >
    <div
      className="ant-radio-group ant-radio-group-outline"
    >
      <label
        className="ant-radio-wrapper ant-radio-wrapper-checked"
      >
        <span
          className="ant-radio ant-radio-checked"
          style={Object {}}
        >
          <input
            checked={true}
            className="ant-radio-input"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            type="radio"
            value={10}
          />
          <span
            className="ant-radio-inner"
          />
        </span>
        <span>
          10
        </span>
      </label>
      <label
        className="ant-radio-wrapper"
      >
        <span
          className="ant-radio"
          style={Object {}}
        >
          <input
            checked={false}
            className="ant-radio-input"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            type="radio"
            value={20}
          />
          <span
            className="ant-radio-inner"
          />
        </span>
        <span>
          20
        </span>
      </label>
      <label
        className="ant-radio-wrapper"
      >
        <span
          className="ant-radio"
          style={Object {}}
        >
          <input
            checked={false}
            className="ant-radio-input"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            type="radio"
            value={30}
          />
          <span
            className="ant-radio-inner"
          />
        </span>
        <span>
          30
        </span>
      </label>
      <label
        className="ant-radio-wrapper"
      >
        <span
          className="ant-radio"
          style={Object {}}
        >
          <input
            checked={false}
            className="ant-radio-input"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            type="radio"
            value={50}
          />
          <span
            className="ant-radio-inner"
          />
        </span>
        <span>
          50
        </span>
      </label>
      <label
        className="ant-radio-wrapper"
      >
        <span
          className="ant-radio"
          style={Object {}}
        >
          <input
            checked={false}
            className="ant-radio-input"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            type="radio"
            value={100}
          />
          <span
            className="ant-radio-inner"
          />
        </span>
        <span>
          100
        </span>
      </label>
    </div>
  </div>
  <button
    className="c0"
    onClick={[Function]}
    onMouseDown={[Function]}
    onTouchStart={[Function]}
  >
    Pay
  </button>
</div>
`;
